x-pack/plugins/observability/public/pages/alerts/workflow_status_filter.tsx

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiButtonGroup, EuiButtonGroupOptionProps } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+import type { AlertWorkflowStatus } from '../../../common/typings';
+
+export interface WorkflowStatusFilterProps {
+  status: AlertWorkflowStatus;
+  onChange: (value: AlertWorkflowStatus) => void;
+}
+
+const options: EuiButtonGroupOptionProps[] = [
+  {
+    id: 'open',
+    label: i18n.translate('xpack.observability.alerts.workflowStatusFilter.openButtonLabel', {
+      defaultMessage: 'Open',
+    }),
+    'data-test-subj': 'WorkflowStatusFilter open button',
+  },
+  {
+    id: 'acknowledged',
+    label: i18n.translate(
+      'xpack.observability.alerts.workflowStatusFilter.acknowledgedButtonLabel',
+      {
+        defaultMessage: 'Acknowledged',
+      }
+    ),
+    'data-test-subj': 'WorkflowStatusFilter acknowledged button',
+  },
+  {
+    id: 'closed',
+    label: i18n.translate('xpack.observability.alerts.workflowStatusFilter.closedButtonLabel', {
+      defaultMessage: 'Closed',
+    }),
+    'data-test-subj': 'WorkflowStatusFilter closed button',
+  },
+];
+
+export function WorkflowStatusFilter({ status = 'open', onChange }: WorkflowStatusFilterProps) {
+  return (
+    <EuiButtonGroup
+      legend="Filter by"
+      color="primary"
+      options={options}
+      idSelected={status}
+      onChange={(id) => onChange(id as AlertWorkflowStatus)}
```

Comment:
The possible `id` values come from the `options` above. I think a better option is constrain the typing of `options` to make sure we never use an invalid ID instead of adding a runtime check. What do you think?